pymatgen/core/structure.py

```diff
@@ -644,31 +644,40 @@ def extract_cluster(self, target_sites: list[Site], **kwargs) -> list[Site]:
             others = new_others
         return cluster
 
-    def _calculate(
-        self, struct: Structure | Molecule, calculator: str | Calculator, verbose: bool = False
-    ) -> Structure | Molecule:
+    def _calculate(self, calculator: str | Calculator, verbose: bool = False) -> Structure | Molecule:
         """
         Performs an ASE calculation.
 
         Args:
-            struct: Structure or Molecule to run ASE calculation on.
-            calculator: An ASE Calculator or a string from the following options: "m3gnet",
-                "gfn2-xtb".
+            calculator (str | Calculator): An ASE Calculator or a string from the following case-insensitive
+                options: "m3gnet", "gfn2-xtb", "chgnet".
             verbose (bool): whether to print stdout. Defaults to False.
+                Has no effect when calculator=='chgnet'.
 
         Returns:
-            Structure: Structure or Molelcule following ASE calculation.
+            Structure | Molecule: Structure or Molecule with new calc attribute containing
+                result of ASE calculation.
         """
+        if isinstance(calculator, str) and calculator.lower() == "chgnet":
+            from chgnet.model import CHGNet
+
+            chgnet = CHGNet.load()
+            preds = chgnet.predict_structure(self)
+
+            out = self.copy()  # type: ignore[attr-defined]
+            out.calc = preds
```

Comment:
@arosen93 @shyuep Setting `out.calc = preds` is inconsistent with the other `calculator` options which store a `ase.Calculator` instance as the `calc` attribute on the calculated structure. But not sure this is the optimal API design and feels like fitting a round peg in a square hole to try and conform CHGNet to this which just returns a dict with predicted `energies`, `forces`, `stresses` and `magmoms`.

Reply:
In general, I think the mechanism to return calculated results (beyond the updated `Structure`) could/should be improved, so I wouldn't worry about trying to match what exists currently if you have a better idea. My thinking at the time was that, originally, the updated `Structure` object was the only thing returned. But the user also calculated forces, stresses, magmoms, etc. so it seemed worthwhile to return those too. Rather than just sticking the calculator (which contains the results in `calc.results`) onto the `Structure` object, it might be cleaner to add a kwarg that would enable the results to be returned as a separate `return` variable. Or maybe there is another clean way to do it. But the tl;dr here is that if there's a way to make it more consistent/cleaner, I would suggest doing so because that was bothering me when I wrote it.

Reply:
But couldn't one then just use atomate2? There, you get everything and it also only takes a few lines of code. Import the job, run it with a memory db and then access the output.

Reply:
That's true (although I don't necessarily view the users of Pymatgen to necessarily be the same as those who would open up Atomate2).

Since this is just meant to be a "quick cleanup" kind of thing, one could quite reasonably argue that you're just asking for the `Structure` or `Molecule` to update if you're doing `.relax()`.

In summary, I'm sure you can come to a clean resolution, @janosh!

Reply:
You can also always use the potentials itself in combination with ase. Thus, I don't really see why anyone would use the pymatgen implementation to access the forces etc.

I would see it as a convenient method to clean up a structure before creating a quantum chemistry input, for example.

Reply:
I agree. Sometimes less is more.